src/averageAges.js

```diff
@@ -55,7 +74,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(
+    women => people.map(person => person.mother).includes(women.name)
+  );
+
+  const children = people.filter(
+    person => mothers.map(mother => mother.name).includes(person.mother)
+  );
+
+  const validChildren = onlyWithSon
+    ? children.filter(child => child.sex === 'm')
+    : children;
```

Comment:
This can be done in one filter call. You need to do only the necessary operations.